client/Components/AdminStretches/AdminStretches.js

```diff
@@ -36,72 +50,185 @@ class AdminStretches extends Component {
     console.log(searchTerm)
   }
 
+  //filter by category tick box
+  selectCategoryFilter = ({ target }) => {
+    let { filterCategoryNames } = this.state
+    if (filterCategoryNames.includes(target.value)) {
+      filterCategoryNames = filterCategoryNames.filter(cat => cat !== target.value)
+    } else {
+      filterCategoryNames.push(target.value)
+    }
+    this.setState({ filterCategoryNames })
+  }
+
+  // filter by selected category
+  applyCategoryFilter = evt => {
+    evt.preventDefault()
+    const { filterCategoryNames } = this.state
+    let { stretches, filterStretches, fetchStretches } = this.props
+    if (filterCategoryNames.length === 0) {
+      fetchStretches()
+    } else {
+      const newStretches = stretches.filter(stretch => filterCategoryNames.includes(stretch.categoryName))
+      filterStretches(newStretches)
+    }
+  }
+
+  // clear filter
+  clearCategoryFilter = evt => {
+    evt.preventDefault()
+    const { fetchStretches } = this.props
+    document.querySelectorAll('input[type=checkbox]').forEach(el => {
+      el.checked = false;
+    })
+    this.setState({ filterCategoryNames: [] })
+    fetchStretches()
+  }
+
+  //filter by author
+  selectAuthorFilter = ({ target }) => {
+    this.setState({ selectedAuthor: target.value })
+  }
+
+  // filter by selected author
+  applyAuthorFilter = evt => {
+    evt.preventDefault()
+    const { selectedAuthor } = this.state
+    let { stretches, filterStretches } = this.props
+    const newStretches = stretches.filter(stretch => stretch.authorName === selectedAuthor)
+    filterStretches(newStretches)
+  }
+
+  // clear filter
+  clearAuthorFilter = evt => {
+    evt.preventDefault()
+    const { fetchStretches } = this.props
+    this.setState({ selectedAuthor: '' })
+    fetchStretches()
+  }
+
   render() {
     const { searchTerm } = this.state
-    const { stretches, categories } = this.props
+    const { stretches } = this.props
+
+    const allCategories = []
+    stretches.map(stretch => allCategories.push(stretch.categoryName))
+    const categories = [...new Set(allCategories)]
+
```

Comment:
I think we are loading the categories on the top level so you probably safe grabbing the categories from the state